IndexedDB/remove-listener-in-handler.any.js

```diff
@@ -0,0 +1,43 @@
+// META: script=support-promises.js
+
+const DB = databaseName("");
+
+const STORE = "store";
+
+// In the bug tickled by this test case, only two of the queries
+// worked and the rest stalled! Do five iterations just to make sure.
+const ITERATIONS = 5;
+
+const t = async_test("removeEventListener in a handler doesn't interfere with future transactions.");
+
+t.add_cleanup(() => {
+    indexedDB.deleteDatabase(DB);
+})
+
+let count = 0;
+
+const go = () => {
+    const dbReq = indexedDB.open(DB, 1);
+    dbReq.addEventListener("upgradeneeded", t.step_func((ev) => {
+        ev.target.result.createObjectStore(STORE, { autoIncrement: true });
+    }));
+    dbReq.addEventListener("success", t.step_func((ev) => {
+        const db = ev.target.result
+        const tx = db.transaction(STORE, "readwrite");
+        const complete = t.step_func(() => {
+            tx.removeEventListener('complete', complete);
+        });
+        tx.addEventListener('complete', complete);
+        const store = tx.objectStore(STORE);
+        store.get(0).addEventListener("success", t.step_func(() => {
+            count += 1;
+            if (count === ITERATIONS) {
+                t.done();
```

Comment:
Since there are no assertions, maybe add a comment e.g. "Test passes if all iterations complete, fails if it times out."

This is implied by the comment up where `ITERATIONS` is defined, but I also had to read https://github.com/dumbmatter/fakeIndexedDB/issues/26 to confirm. Links or comments in the test to help future maintainers understand what this test is verifying (i.e. that it's a regression test for a previously buggy implementation)  would be great. Otherwise, some poor maintainer will try and figure out what part of the spec this is validating.

Reply:
I think the comment at the top is good enough now - it explicitly says that passing all the iterations is good enough, and the file's short enough that it's unlikely to be missed.